src/analysis/retail/rogue/outlaw/modules/talents/BladeRush.tsx

```diff
@@ -1,10 +1,28 @@
 import RESOURCE_TYPES, { getResource } from 'game/RESOURCE_TYPES';
 import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
-import Events, { CastEvent } from 'parser/core/Events';
+import Events, {
+  CastEvent,
+  ResourceChangeEvent,
+  UpdateSpellUsableEvent,
+  UpdateSpellUsableType,
+} from 'parser/core/Events';
 import Abilities from 'parser/core/modules/Abilities';
 import SpellUsable from 'parser/shared/modules/SpellUsable';
-import SPELLS from 'common/SPELLS';
 import TALENTS from 'common/TALENTS/rogue';
+import { SpellLink } from 'interface';
+import { RoundedPanel } from 'interface/guide/components/GuideDivs';
+import { Uptime } from 'parser/ui/UptimeBar';
+import uptimeBarSubStatistic from 'parser/ui/UptimeBarSubStatistic';
+import { explanationAndDataSubsection } from 'interface/guide/components/ExplanationRow';
+import { BoxRowEntry, PerformanceBoxRow } from 'interface/guide/components/PerformanceBoxRow';
+import { QualitativePerformance } from 'parser/ui/QualitativePerformance';
+
+const MIN_ENERGY_TRESHOLD = 70;
+const MAX_ENERGY_TRESHOLD = 90;
+const ACCEPTABLE_ENERGY_TRESHOLD = 80;
+
+
+//-- TODO: MAX_ENERGY_TRESHOLD should scale with targets hit to support aoe fights
```

Comment:
It's somewhat difficult to get a good handle on whether or not a fight is a good candidate for AoE solely based on logs. For instance, if you're assigned to be boss group on Broodkeeper Diurna, you likely won't be doing any AoE damage despite the fight's potential. On Eranog, you have very short periods of high burst AoE.

It's not to say that it's impossible, but there are likely lower hanging fruit that will provide more value.

Reply:
Maybe this comment was missunderstood so let me explain it a bit better, blade rush basically gain value with target count, so let say the threshold in single target would be 70 energy, with 5 targets it would increase to be much higher to the point you would almost want to press the spell on cd.
So the idea behind this was to check the numbers of targets around the user to implement that as a factor to the Max_energy_treshold.

I'm not entirely sure if we can check the numbers of targets around us tho? But at the very least outlaw has blade flurry constantly running in aoe so I could check the number of target hit.